analysers/Analyser_Merge_Geocode_City_CSV.py

```diff
@@ -0,0 +1,77 @@
+#!/usr/bin/env python
+#-*- coding: utf-8 -*-
+
+###########################################################################
+##                                                                       ##
+## Copyrights François Lacombe 2022                                      ##
+##                                                                       ##
+## This program is free software: you can redistribute it and/or modify  ##
+## it under the terms of the GNU General Public License as published by  ##
+## the Free Software Foundation, either version 3 of the License, or     ##
+## (at your option) any later version.                                   ##
+##                                                                       ##
+## This program is distributed in the hope that it will be useful,       ##
+## but WITHOUT ANY WARRANTY; without even the implied warranty of        ##
+## MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the         ##
+## GNU General Public License for more details.                          ##
+##                                                                       ##
+## You should have received a copy of the GNU General Public License     ##
+## along with this program.  If not, see <http://www.gnu.org/licenses/>. ##
+##                                                                       ##
+## This tool uses French administrative API to find center coordinates   ##
+## of cities                                                             ##
+## See https://geo.api.gouv.fr/decoupage-administratif                   ##
+##                                                                       ##
+###########################################################################
+
+from .Analyser_Merge import Source
+from modules import downloader
+import json
+
+
+class Geocode_City_CSV(Source):
+
+    def __init__(self, source, logger, citycode = None, delimiter = ',', encoding = 'utf-8'):
+        self.source = source
+        self.delimiter = delimiter
+        self.citycode = citycode
+        self.encoding = encoding
+        self.logger = logger
+
+    def __getattr__(self, name):
+        return getattr(self.source, name)
+
+    def open(self):
+        return open(downloader.update_cache('citycoded://' + self.source.fileUrl, 60, fetch=self.fetch))
```

Comment:
@frodrigo can you confirm this is appropriate to cache geocoding results, right?

Geocoded files may have several lines involving a given city.
I expect the API will be queried once and then use cached results for further records.
I'm not sure how cache is handled exactly.

Reply:
No. It is to cache the result of the geocoding.

Reply:
So for several queries `GET https://geo.api.gouv.fr/communes/74010?fields=centre&format=json&geometry=centre` the API will be queried only once, right?

Reply:
By the result I mean the global CSV. The output of the processing.

If you want to cache query you can use https://github.com/osm-fr/osmose-backend/blob/master/modules/downloader.py#L140 or one of previous in the module.

Reply:
Got it, thanks.

I've edited the code to use urlread instead of Python requests.Session.